rlclientlib/model_mgmt/restapi_data_transport.cc

```diff
@@ -39,14 +43,23 @@ namespace reinforcement_learning { namespace model_management {
    */
 
   int restapi_data_transport::get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status) {
-
+    // Get request URI and start the request.
+    http_request request(methods::HEAD);
+    request.headers() = _header;
     // Build request URI and start the request.
-    auto request_task = _httpcli->request(methods::HEAD)
-      // Handle response headers arriving.
-      .then([&](http_response response) {
-      if ( response.status_code() != 200 )
-        RETURN_ERROR_ARG(_trace, status, http_bad_status_code, _httpcli->get_url());
-
+    auto request_task = _httpcli->request(request).then([&](http_response response) {
+      if ( response.status_code() != 200)
+      {
+        if ( response.status_code() == 404)
+        {
+          RETURN_IF_FAIL(get_data_info_using_get(response, status));

```

Comment:
Why immediately call get if head fails?

Reply:
while using modelBlobUri it has head api call where as /model api in frontEnd doesn't have HEAD call at the moment. Instead if 404 is received we will give it a try with GET method